gobblin-runtime/src/main/java/org/apache/gobblin/runtime/JobContext.java

```diff
@@ -146,12 +152,23 @@ public JobContext(Properties jobProps, Logger logger, SharedResourcesBroker<Gobb
 
     this.datasetStateStore = createStateStore(ConfigUtils.propertiesToConfig(jobProps));
     this.jobHistoryStoreOptional = createJobHistoryStore(jobProps);
+    this.globalWatermarkDatasetUrn = jobProps.containsKey(ConfigurationKeys.GLOBAL_WATERMARK_DATASET_URN) ? Optional
```

Comment:
Why is this part of `JobContext`, what is the intended behavior as a framework level flag?